pkg/operator/workload/manage.go

```diff
@@ -0,0 +1,288 @@
+package workload
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"reflect"
+	"strings"
+
+	"github.com/openshift/cluster-openshift-controller-manager-operator/pkg/operator/v311_00_assets"
+	"github.com/openshift/cluster-openshift-controller-manager-operator/pkg/util"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"github.com/openshift/library-go/pkg/operator/resource/resourcehash"
+	"github.com/openshift/library-go/pkg/operator/resource/resourcemerge"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceread"
+	"k8s.io/client-go/kubernetes"
+
+	operatorapiv1 "github.com/openshift/api/operator/v1"
+	proxyclientv1 "github.com/openshift/client-go/config/listers/config/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	appsclientv1 "k8s.io/client-go/kubernetes/typed/apps/v1"
+	coreclientv1 "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+type manageConfigMapFn func(
+	kubernetes.Interface,
+	coreclientv1.ConfigMapsGetter,
+	events.Recorder,
+	*operatorapiv1.OpenShiftControllerManager,
+) (*corev1.ConfigMap, bool, error)
+
+type manageClientCAFn func(
+	client coreclientv1.CoreV1Interface,
+	recorder events.Recorder,
+) (bool, error)
+
+func manageOpenShiftControllerManagerClientCA_v311_00_to_latest(
```

Comment:
We should drop references to v311 everywhere. This is an artifact of the very early OCP 4 days when direct upgrades from v3.11 to v4.x was on the table.

Note - perhaps should be done in a separate PR.

Reply:
Since the functions containing `v311` are coming from the original version, I think we should make the renaming in a subsequent PR is a good idea.